docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,213 @@
+Optimization Guide
+==================
+
+Here you can find more information about Modin's optimizations both for a pipeline of operations as well as for specific operations.
+
+
+Understanding Modin's partitioning mechanism
+"""""""""""""""""""""""""""""""""""""""""""""""
+Modin's partitioning is crucial for performance; so we recommend expert users to understand Modin's
+partitioning mechanism and how to tune it, to achieve better performance.
+
+How does Modin partition a dataframe
+------------------------------------
+
+Modin uses a partitioning scheme that partitions a dataframe along both axes, resulting in a matrix
+of partitions. The row and column chunk sizes are computed independently based
+on the length of the appropriate axis and Modin's special :doc:`configuration variables </flow/modin/config>`
+(``NPartitions`` and ``MinPartitionSize``):
+
+- `NPartitions` is the maximum number of splits along an axis; by default, it equals the number of cores on the machine.
+- `MinPartitionSize` is the minimum number of rows/columns to do a split. For instance, if `MinPartitionSize`
+  is 32, the column axis will not be split unless the amount of columns is greater than 32. If it is is greater, for example, 34, 
+  then the column axis is sliced into two partitions: containing 32 and 2 columns accordingly.
```

Comment:
Possible, need to add small mentioning about a case when data shape is couldn't be split to `NPartitions` splits because of `MinPartitionSize` restrictions. For example, what will happen in case shape is (70, 70) and `NParttions` is 112.

Reply:
Shown in the picture below.